File: capacitor-sora-editor-1.2.1/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/smartEnter/NewlineHandleResult.java

package io.github.abc15018045126.sora.lang.smartEnter;

public class NewlineHandleResult {

    /**
     * Text to insert
     */
    public final CharSequence text;

    /**
     * Count to shift left from the end of {@link NewlineHandleResult#text}
     */
    public final int shiftLeft;

    public NewlineHandleResult(CharSequence text, int shiftLeft) {
        this.text = text;
        this.shiftLeft = shiftLeft;
        if (shiftLeft < 0 || shiftLeft > text.length()) {
            throw new IllegalArgumentException("invalid shiftLeft");
        }
    }

}

File: android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/smartEnter/NewlineHandleResult.kt
package io.github.abc15018045126.sora.lang.smartEnter

class NewlineHandleResult(

    @JvmField val text: CharSequence,

    @JvmField val shiftLeft: Int
) {
    init {
        if (shiftLeft < 0 || shiftLeft > text.length) {
            throw IllegalArgumentException("invalid shiftLeft")
        }
    }
}
